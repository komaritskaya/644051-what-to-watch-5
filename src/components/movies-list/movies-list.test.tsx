import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list';

it(`Should Movies List component render correctly`, () => {
  const movies = [
    {
      id: `123`,
      title: `Some Film`,
      poster: `some-poster.jpg`,
      genre: `comedy`,
      year: 2000,
      director: `Some Director`,
      cast: [`Actor One`, `Actor Two`],
      cover: `some-bg-poster.jpg`,
      description: `Damn good film`,
      rating: 10.0,
      reviewsCount: 100,
    },
    {
      id: `456`,
      title: `Another Film`,
      poster: `another-poster.jpg`,
      genre: `drama`,
      year: 2000,
      director: `Another Director`,
      cast: [`Actor One`, `Actor Two`],
      cover: `another-bg-poster.jpg`,
      description: `Awful film`,
      rating: 2.0,
      reviewsCount: 50,
    },
  ];

  const tree = renderer
    .create(<MoviesList
      movies={movies}
      onCardClick={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
