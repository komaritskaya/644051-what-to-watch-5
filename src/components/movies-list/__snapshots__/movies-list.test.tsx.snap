// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Movies List component render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        alt="Some Film"
        height="175"
        src="img/some-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Some Film
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        alt="Another Film"
        height="175"
        src="img/another-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Another Film
      </a>
    </h3>
  </article>
</div>
`;
